client/components/platform-checkout/save-user/phone-number-input.js

```diff
@@ -1,144 +1,111 @@
 /**
  * External dependencies
  */
-import React, { useEffect, useState } from 'react';
+import React, { useEffect, useState, useRef } from 'react';
 import { __ } from '@wordpress/i18n';
 import intlTelInput from 'intl-tel-input';
+import './style.scss';
 
 /**
  * Internal dependencies
  */
 // eslint-disable-next-line import/no-unresolved
 import utils from 'iti/utils';
 
-const PhoneNumberInput = ( { handlePhoneNumberChange } ) => {
-	const [ inputValue, setInputValue ] = useState(
-		document.getElementById( 'billing_phone' )?.value ?? ''
-	);
+const PhoneNumberInput = ( {
+	onValueChange,
+	value,
+	onValidationChange = () => {},
+	inputProps = {},
+} ) => {
 	const [ inputInstance, setInputInstance ] = useState( null );
 	const [ isValid, setIsValid ] = useState( true );
```

Comment:
We now have 2 state variables indicating whether the phone number is valid or not, one in the parent component and another in this. How about we consolidate the variables and have just one?